src/models/StyledComponent.js

```diff
@@ -58,8 +58,102 @@ class BaseStyledComponent extends Component<*> {
   static defaultProps: Object
   static warnTooManyClasses: Function
 
+  renderOuter: Function
+  renderInner: Function
+  styleSheet: ?StyleSheet
+
   attrs = {}
 
+  constructor() {
+    super()
+    this.renderOuter = this.renderOuter.bind(this)
+    this.renderInner = this.renderInner.bind(this)
+  }
+
+  render() {
+    return <StyleSheetConsumer>{this.renderOuter}</StyleSheetConsumer>
+  }
+
+  renderOuter(styleSheet?: StyleSheet) {
+    this.styleSheet = styleSheet
```

Comment:
Does this need an `if (!this.styleSheet) this.styleSheet = styleSheet`?

Reply:
Nope, we want to continually update it. This is just a place to stash the variable so we can reference it in `renderInner` without making new functions every render.